net-libs/grpc/grpc-1.62.1.ebuild

```diff
@@ -0,0 +1,210 @@
+	local rstring="${_}"
+	sed -i -E "s/( +)gtest/\1${rstring}/g" "CMakeLists.txt" || die
+
+	# Integrate tests with ctest rather than the custom test framework.
+	# Formatted with dev-python/black.
+	"${EPYTHON}" - <<-EOF | tee -a "CMakeLists.txt"
+		import json, pathlib
+
+		print("if(gRPC_BUILD_TESTS)")
+		print("  ENABLE_TESTING()")
+		for line in [
+		  json.dumps([t["name"], "./" + t["name"], *t["args"]]).translate(
+		  str.maketrans(dict.fromkeys("[],", None))
+		  )
+		  for t in json.loads(
+		    pathlib.Path("tools/run_tests/generated/tests.json").read_text()
+		  )
+		  if "linux" in t["platforms"] and not t["flaky"] and not t.get("boringssl", False)
+		]:
+		  print(f"  add_test({line})")
+		print("endif()")
+	EOF
+
+	# Weird path issue.  All tests except these two assume they are running from top-level src
+	sed -i -E "s/lslash != nullptr/false/" "test/core/http/httpcli_test_util.cc" || die
+
+	# Called via system() by some of the C++ sources, respect EPYTHON
+	sed -i -E "s#for p in #for p in \"${EPYTHON}\"#" "tools/distrib/python_wrapper.sh" || die
+
+	# Respect EPYTHON when testing, don't touch installed files otherwise
+	use test && python_fix_shebang --force "${S}"
+
+	cmake_src_prepare
+
+	# un-hardcode libdir
+	sed -i "s@/lib@/$(get_libdir)@" cmake/pkg-config-template.pc.in || die
+
+	# suppress network access, package builds fine without the submodules
+	mkdir "${S}/third_party/opencensus-proto/src" || die
+
+	soversion_check
+}
+
+src_configure() {
+	# https://github.com/grpc/grpc/issues/29652
+	filter-lto
+
+	local mycmakeargs=(
+		-DgRPC_INSTALL=ON
+		-DgRPC_ABSL_PROVIDER=package
+		-DgRPC_CARES_PROVIDER=package
+		-DgRPC_INSTALL_CMAKEDIR="$(get_libdir)/cmake/${PN}"
+		-DgRPC_INSTALL_LIBDIR="$(get_libdir)"
+		-DgRPC_PROTOBUF_PROVIDER=package
+		-DgRPC_RE2_PROVIDER=package
+		-DgRPC_SSL_PROVIDER=package
+		-DgRPC_ZLIB_PROVIDER=package
+		-DgRPC_BUILD_TESTS=$(usex test)
+		-DCMAKE_CXX_STANDARD=17
+		-DCMAKE_DISABLE_FIND_PACKAGE_systemd=$(usex !systemd)
+		$(usex test '-DgRPC_BENCHMARK_PROVIDER=package' '')
+	)
+	cmake_src_configure
+}
+
+src_test() {
+	# This is normally done with start_port_server.py, but this forks and exits,
+	# while we need to capture the pid, so run it ourselves
+	"${EPYTHON}" "tools/run_tests/python_utils/port_server.py" \
+		-p 32766 -l "${T}/port_server.log" &
+	local port_server_pid="${!}"
+
+	# Reimplementation of what start_port_server.py does with curl
+	curl --retry 9999 --retry-all-errors --retry-max-time 120 \
+		--fail --silent --output /dev/null "http://localhost:32766/get" || die
+
+	local disabled_tests=(
+		-E no_logging_test # hangs everywhere, no output
+		-E grpc_tool_test # fails everywhere
+		-E examine_stack_test # fails on amd64 only
+		-E stack_tracer_test # fails on amd64 only
+		-E endpoint_pair_test # fails on alpha
+		-E event_poller_posix_test # fails on alpha
+		-E tcp_posix_test # fails on alpha
+	)
+
+	cmake_src_test "${disabled_tests[@]}"
```

Comment:
I don't know why, but it looks like it didn't work (the disable)?
Ran it on my amd64 testing container, and it failed on those expected tests, and hanged on `no_logging_test`. Maybe you need to combine them into single regex? Since I indeed don't see `tcp_posix_test` in logs (the last exclude arg). Something like:
`-E "(no_logging_test|grpc_tool_test|examine_stack_test)"`